x/concentrated-liquidity/position_test.go

```diff
@@ -130,21 +196,71 @@ func (s *KeeperTestSuite) TestInitOrUpdatePosition() {
+			// ---Tests for ensuring uptime accumulators behaved as expected---
+
+			// Get updated accumulators and accum values
+			newUptimeAccums, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(s.Ctx, test.param.poolId)
+			s.Require().NoError(err)
+			newUptimeAccumValues, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulatorValues(s.Ctx, test.param.poolId)
 			s.Require().NoError(err)
 
-			// If frozen for more than a specific uptime's period, the record should exist
-			for uptimeIndex, uptime := range types.SupportedUptimes {
-				recordExists, err := uptimeAccums[uptimeIndex].HasPosition(positionName)
+			// Setup for checks
+			actualUptimeAccumDelta, expectedUptimeAccumValueGrowth, expectedIncentiveRecords, expectedGrowthCurAccum := emptyAccumValues, emptyAccumValues, test.incentiveRecords, sdk.DecCoins{}
+			freezePeriod := test.param.frozenUntil.Sub(s.Ctx.BlockTime())
+			timeElapsedSec := sdk.NewDec(int64(test.timeElapsedSinceInit)).Quo(sdk.NewDec(10e8))
+			positionName := string(types.KeyFullPosition(validPoolId, s.TestAccs[0], test.param.lowerTick, test.param.upperTick, test.param.frozenUntil))
+
+			// Loop through each supported uptime for pool and ensure that:
+			// 1. Position is properly updated on it
+			// 2. Accum value has changed by the correct amount
+			for uptimeIndex, uptime := range supportedUptimes {
+
+				// Position-related checks
+
+				// If frozen for more than a specific uptime's period, the record should exist
+				recordExists, err := newUptimeAccums[uptimeIndex].HasPosition(positionName)
 				s.Require().NoError(err)
-				if test.param.frozenUntil.Sub(s.Ctx.BlockTime()) >= uptime {
+				if freezePeriod >= uptime {
 					s.Require().True(recordExists)
+
+					// Ensure position's record has correct values
+					positionRecord, err := accum.GetPosition(newUptimeAccums[uptimeIndex], positionName)
+					s.Require().NoError(err)
+					s.Require().Equal(newUptimeAccums[uptimeIndex].GetValue(), positionRecord.InitAccumValue)
+					s.Require().Equal(test.expectedLiquidity, positionRecord.NumShares)
 				} else {
 					s.Require().False(recordExists)
 				}
+
+				// Accumulator value related checks
+
+				if test.positionExists {
+					// Track how much the current uptime accum has grown by
+					actualUptimeAccumDelta[uptimeIndex] = newUptimeAccumValues[uptimeIndex].Sub(initUptimeAccumValues[uptimeIndex])
+					if timeElapsedSec.GT(sdk.ZeroDec()) {
+						expectedGrowthCurAccum, expectedIncentiveRecords, err = cl.CalcAccruedIncentivesForAccum(s.Ctx, uptime, test.param.liquidityDelta, timeElapsedSec, expectedIncentiveRecords)
+						s.Require().NoError(err)
+						expectedUptimeAccumValueGrowth[uptimeIndex] = expectedGrowthCurAccum
+					}
+				} else {
+					// if no position init, should remain empty
+					s.Require().NoError(err)
+					s.Require().Equal(initUptimeAccumValues[uptimeIndex], newUptimeAccumValues[uptimeIndex])
+				}
```

Comment:
Slightly confused about this else statement. This is checking the response of `HasPosition` right? Didn't we already check `NoError` right after calling `HasPosition`? I think we just need to check `Equal`

Reply:
Good point, removed